homework-08/main.js

```diff
@@ -0,0 +1,142 @@
+// 1 Створити масив, довжину та елементи якого задає користувач (через prompt). Елементами масиву повинні бути числа.
+const numbers = [];
+while (true) {
+    let value = prompt('Enter numbers ');
+        if ( value === '' || value === null || value === undefined ) {
+            break;
+        }    
+        if(isNaN(value)) {
+            console.log(`${value} is not a number`)
+        } 
+        else {
+            numbers.push(+value);
+        }
+};
+console.log(numbers);
+
+// 2 Відсортувати масив за зростанням.
+let sortedNumbers = numbers.sort((a, b) => a - b);
+console.log(sortedNumbers);
+
+// 3
+const PRODUCT_ARRAY = [ 
+    {productName: 'bread', productPrice: 14.5, productQuantity: 2},
+    {productName: 'butter', productPrice:  25, productQuantity: 1},
+    {productName: 'milk', productPrice:  17, productQuantity: 3},
+    {productName: 'flour', productPrice:  12.75, productQuantity: 1},
+    {productName: 'sugar', productPrice:  35, productQuantity: 5},
+    {productName: 'salt', productPrice:  10, productQuantity: 4},
+    {productName: 'oil', productPrice:  45, productQuantity: 7},
+    {productName: 'vinegar', productPrice:  10, productQuantity: 1},
+    {productName: 'nuts', productPrice:  18, productQuantity: 6},
+]
+
+// Порахувати загальну вартість корзини та вивести суму у зрозумілому форматі
+const totalProductPrice = PRODUCT_ARRAY.reduce(
+    (accumulator, item) => {
+        return accumulator + item.productPrice * item.productQuantity;    
+    },
+0);
+
+console.log(`The total price of products is ${totalProductPrice}`);
+
+// Порахувати найменшу кількість продукту, який потрібно купити
+const result = PRODUCT_ARRAY.reduce((minValue, curValue) => 
+    curValue.productQuantity < minValue ? curValue.productQuantity : minValue
+, Infinity); 

```

Comment:
Infinity is a convenient value to use there, in the first step of the loop it will be changed to our data values, but at the same time the Infinity value is guaranteed not to exist in our data